### PR TITLE
Add registry metadata to published components.

### PR DESCRIPTION
**Pull request by peterhuene**
This PR adds registry metadata to components published to a registry with
`cargo component publish` and `wit publish`.

### CODE REVIEW

File: crates/wit/src/lib.rs [hunk 1]
```diff
@@ -273,6 +274,56 @@ struct PublishOptions<'a> {
...
+    }
+
+    if !config.categories.is_empty() {
+        metadata.set_categories(Some(config.categories.clone()));
+    }
+
+    metadata.set_description(config.description.clone());
+
+    // TODO: registry metadata should have keywords
+    // if !package.keywords.is_empty() {
```
Comment (fibonacci1729) [inline crates/wit/src/lib.rs:290]: Are `keywords` distinct from `categories`? They seem coupled in the original registry discussion.

File: crates/wit/src/commands/add.rs [hunk 2]
```diff
@@ -127,16 +123,6 @@ impl AddCommand {
                     .dependencies
                     .insert(id.clone(), Dependency::Package(package));
 
-                // Resolve all dependencies to ensure that the lockfile is up to date.
-                resolve_dependencies(
```
Comment (fibonacci1729) [inline crates/wit/src/commands/add.rs:131]: I am sure i am missing something, but why is this removed?

**Review by fibonacci1729 — APPROVED** [review]

**Comment by fibonacci1729** [comment]
LGTM! Just a couple question to educate myself.

File: crates/wit/src/lib.rs (hunk 1 above)
Comment (peterhuene) [reply crates/wit/src/lib.rs:290]: `crates.io` has them as [distinct fields](https://doc.rust-lang.org/cargo/reference/manifest.html#the-keywords-field), where keywords is for describing the package and the category with a set of predefined [slugs](https://crates.io/category_slugs) to use.

I wasn't following the discussion around the metadata implementation closely, so it sounds like we should be populating this off of `keywords` instead of categories or perhaps both?

File: crates/wit/src/commands/add.rs (hunk 2 above)
Comment (peterhuene) [reply crates/wit/src/commands/add.rs:131]: This was to fix a double-resolution that was taking place, once to resolve the package being added and then again to resolve all packages in an attempt to update the lock file.

Updating the lock file isn't strictly necessary from the command, so this was removed to improve the UX.

Comment (fibonacci1729) [reply crates/wit/src/commands/add.rs:131]: Awesome, makes complete sense. Thanks for the explanation.

File: crates/wit/src/lib.rs (hunk 1 above)
Comment (peterhuene) [reply crates/wit/src/lib.rs:290]: I'm going to switch to populating the `categories` field from the cargo `keywords` as I believe that's more of the intention of this field; the predefined categories list from cargo doesn't map to what we'd want in a component registry anyway.

I've opened https://github.com/bytecodealliance/registry/issues/197 to track the confusion here.

Comment (peterhuene) [reply crates/wit/src/lib.rs:290]: I've decided to merge this as-is and wait for the resolution on the linked issue; will update `cargo-component` if things change.
